api/envoy/config/listener/v3/listener.proto

```diff
@@ -275,4 +280,13 @@ message Listener {
   // :ref:`use_original_dst <envoy_api_field_config.listener.v3.Listener.use_original_dst>`
   // to true. Default is true.
   google.protobuf.BoolValue bind_to_port = 26;
+
+  // Used to represent an internal listener which does not listen on OSI L4 address but can be used by the
+  // :ref:`envoy cluster <envoy_v3_api_msg_config.cluster.v3.Cluster>` to create a user space connection to.
+  // The internal listener acts as a tcp listener. It supports listener filters and network filter chains.
+  // There are some limitations are derived from the implementation. The known limitations include `connection_balance_config`
+  // is not allowed, and some socket options are not supported.
+  // Require that :ref:`address <envoy_api_field_config.listener.v3.Listener.address>` has field `envoy_internal_address`.
```

Comment:
What happens if multiple listeners bind to the same `server_listener_name`?

Reply:
It should follow the convention as Listener::address field. As per `ListenerManagerImpl`, it should be either NACK or the first-warmed-up win. I don't have a preference though. Does early NACK make more sense?

```
 // The address that the listener should listen on. In general, the address must be unique ...
  core.v3.Address address = 2
```